webhook/resources/backup/validator.go

```diff
@@ -92,3 +98,41 @@ func (b *backupValidator) Create(request *admission.Request, newObj runtime.Obje
 
 	return nil
 }
+
+func (b *backupValidator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) error {
+	oldBackup, ok := oldObj.(*longhorn.Backup)
+	if !ok {
+		return werror.NewInvalidError(fmt.Sprintf("oldObj %v is not a longhorn.Backup", oldObj), "")
+	}
+	newBackup, ok := newObj.(*longhorn.Backup)
+	if !ok {
+		return werror.NewInvalidError(fmt.Sprintf("newObj %v is not a longhorn.Backup", newObj), "")
+	}
+
+	// Allow backup block size mutation only when the existing obj is not set, or correcting the existed invalid value
+	isValidOldBackupBlockSize := b.validateBackupBlockSize(oldBackup, false) == nil
+	if isValidOldBackupBlockSize && oldBackup.Spec.BackupBlockSize != newBackup.Spec.BackupBlockSize {
+		err := fmt.Errorf("changing backup block size for backup %v is not supported", oldBackup.Name)
+		return werror.NewInvalidError(err.Error(), "")
+	}
+	if err := b.validateBackupBlockSize(newBackup, false); err != nil {
+		return werror.NewInvalidError(err.Error(), "")
+	}
+
+	return nil
+}
+
+func (b *backupValidator) validateBackupBlockSize(backup *longhorn.Backup, allowInvalid bool) error {
+	if allowInvalid && backup.Spec.BackupBlockSize == types.BackupBlockSizeInvalid {
+		return nil
+	}
```

Comment:
Add a comment for explaining when the backup block size is BackupBlockSizeInvalid and why we blindly return nil here.